internal/impl/kafka/output_redpanda.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"context"
+	"slices"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	roFieldMaxInFlight = "max_in_flight"
+)
+
+func redpandaOutputConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Beta().
+		Categories("Services").
+		Summary("A Kafka output using the https://github.com/twmb/franz-go[Franz Kafka client library^].").
+		Description(`
+Writes a batch of messages to Kafka brokers and waits for acknowledgement before propagating it back to the input.
```

Comment:
The franz output talks about how it out performs the normal kafka output... Should we give guidance here on using Redpanda vs Franz outputs?

Reply:
I think once we're happy we deprecate the franz components entirely. It's not just performance it's also the general user experience that's improved (pending user feedback).

Reply:
Oh you don't think the ordering will be a performance difference between the Redpanda and kafka_franz components? I thought we'd end up keeping both of those (but removing the old samara based one)